### DISCUSSION TITLE
user requests

### DISCUSSION THREAD

**Discussion by smashrvmco** [iamromulan/cellular-modem-wiki#64] [Ideas/Requests - Quectel AP Repo]
First of all, many many thanks for doing this and getting the info out there for the consumer to use!

I'm sure you guys are trying to implement everything and more than what the InvisaGig toolkit does, with the bandlocking and updates already being implemented / worked on, any plans for a live signal for best placement, and maybe a dark mode for simpleadmin :)

**Comment by iamromulan** [comment]
@dr-dolomite is hard at work on the front end, I know for sure he'll have a dark mode light mode toggle feature soon with the all new Simple admin 2.0

We have been interested in reducing the refresh rate down lower however we have to be careful not to overload the little single core armv7 32-bit processor on board. I believe we can achieve something similar to the IG of a rate of 3 seconds. The new x72 and x75 cards that should be arriving next month from Quectel should be a lot more flexible for things like this.
The RM550V-GL (x72) and RM551E-GL (x75) will have a 64 bit quad core. (Which will complicate the install process of the toolkit installer a little bit, will need to add new checks so the correct binary is downloaded for the correct architecture.)

**Comment by dr-dolomite** [comment]
> First of all, many many thanks for doing this and getting the info out there for the consumer to use!
> 
> I'm sure you guys are trying to implement everything and more than what the InvisaGig toolkit does, with the bandlocking and updates already being implemented / worked on, any plans for a live signal for best placement, and maybe a dark mode for simpleadmin :)

I already implemented dark mode on the Simple Admin 2.0. Stay tuned 😁

**Comment by iamromulan** [comment]
> First of all, many many thanks for doing this and getting the info out there for the consumer to use!
> 
> I'm sure you guys are trying to implement everything and more than what the InvisaGig toolkit does, with the bandlocking and updates already being implemented / worked on, any plans for a live signal for best placement, and maybe a dark mode for simpleadmin :)

Development branch has 2 to 3 seconds stats refresh time now

**Comment by iamromulan** [comment]
Offline deployment requested by 

@sevvian

https://github.com/iamromulan/quectel-rgmii-toolkit/issues/38#issuecomment-2081848427

**Comment by iamromulan** [comment]
@danry25 has requested Automatic/Dynamic band locking based on ping
https://github.com/iamromulan/quectel-rgmii-toolkit/issues/54#issue-2309085875

**Comment by snjzb** [comment]
Hello! I am a developer from China. Can I clone the "quectel-rgmii-toolkit" project for developing a Chinese version?

**Reply by dr-dolomite** [reply]
Im not the maintainer of this repository but I think you can as long as you follow the License Agreement and make sure to include credits where its due.

**Reply by snjzb** [reply]
I'm not very good at maintaining projects on GitHub. I have created an offline version of the installation package, and the included attribution information has not been removed. If needed, I can upload the complete modified files.
![{A37_R` Y )JC4{2Y8I0JIF](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/b09bf519-9aaa-484a-9cd8-ec1cac20e3c3)
![10B~UZE 9RBAMT}BTE2ZFJG](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/eba77c84-0950-4ce8-bda4-44e1c5050121)

**Reply by snjzb** [reply]
The SMS receiving part has been completed. Currently, the SMS sending function can only send messages that do not contain Chinese characters.
![_F7G__}EM{SUZ SMK6QVB7E](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/3f812207-7f55-40e1-85dd-8985bbcee0db)

**Reply by dr-dolomite** [reply]
May I know if how you did it? The send SMS part.

**Reply by snjzb** [reply]
This is the relevant file.
[sms.zip](https://github.com/iamromulan/quectel-rgmii-toolkit/files/15418922/sms.zip)

**Reply by iamromulan** [reply]
> I'm not very good at maintaining projects on GitHub. I have created an offline version of the installation package, and the included attribution information has not been removed. If needed, I can upload the complete modified files.
> ![{A37_R` Y )JC4{2Y8I0JIF](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/b09bf519-9aaa-484a-9cd8-ec1cac20e3c3)
> ![10B~UZE 9RBAMT}BTE2ZFJG](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/eba77c84-0950-4ce8-bda4-44e1c5050121)
> 
> 

This will make it super easy:

https://desktop.github.com/

It's also available for Ubuntu
https://github.com/shiftkey/desktop?tab=readme-ov-file#installation-via-package-manager

@dr-dolomite 
I really think it would be nice to see the option to switch languages on the fly. If need be we can restructure the www folder to allow this.

@snjzb 
I think I remember seeing something about switching character sets with an AT command, maybe that will help enable sending text with Chinese letters. I'll see if I can find it and I'll reply with it. 

For now, your implementation of sending texts with simpleadmin would be a great contribution to the project. If you need help setting up GitHub for collaboration let me know! GitHub desktop will watch the repo folder and allow you to manage changes and pull requests into other repos on GitHub.

**Reply by dr-dolomite** [reply]
Thank you so much. Is it okay if we can also use this?

**Reply by snjzb** [reply]
I tried using the "GitHub Desktop" version. 
I attempted to convert the character set for sending text, but it was not successful. It seems that some configurations are still missing, or it could be a module version issue.
![6DGZZ}Z$(BC)4KMMER9ZB8V](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/f7e4edbf-f464-4f8a-ae4e-06c884347382)

**Reply by dr-dolomite** [reply]
Have you checked smstools3? It looks like they support .wider character set

**Comment by dr-dolomite** [comment]
> This is the relevant file. [sms.zip](https://github.com/iamromulan/quectel-rgmii-toolkit/files/15418922/sms.zip)



> > I'm not very good at maintaining projects on GitHub. I have created an offline version of the installation package, and the included attribution information has not been removed. If needed, I can upload the complete modified files.
> > ![{A37_R` Y )JC4{2Y8I0JIF](https://private-user-images.githubusercontent.com/170070306/333228664-b09bf519-9aaa-484a-9cd8-ec1cac20e3c3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0Nzg1NTQsIm5iZiI6MTcxNjQ3ODI1NCwicGF0aCI6Ii8xNzAwNzAzMDYvMzMzMjI4NjY0LWIwOWJmNTE5LTlhYWEtNDg0YS05Y2Q4LWVjMWNhYzIwZTNjMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyM1QxNTMwNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xYzljYzFkNTYyYWJjZDNjNWNhNzllNjQ2ZTA4M2VhYTcxN2M3ZGMzOTlkN2U3NGUxYzRmZjNjOWE4MDExYzFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.iCMsG-9Tgp2sCszITViLjzWKdNIAqEm8nh3DQSYXlDU)
> > ![10B~UZE 9RBAMT}BTE2ZFJG](https://private-user-images.githubusercontent.com/170070306/333228668-eba77c84-0950-4ce8-bda4-44e1c5050121.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0Nzg1NTQsIm5iZiI6MTcxNjQ3ODI1NCwicGF0aCI6Ii8xNzAwNzAzMDYvMzMzMjI4NjY4LWViYTc3Yzg0LTA5NTAtNGNlOC1iZGE0LTQ0ZTFjNTA1MDEyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyM1QxNTMwNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNzQwMjc0ODAwMTUxNWY1ZmVhOGE3NjRiZWZjMjE4OWNkOGM0NzViNzE1OWYwYzczZDQ2NjljYjkzZTRkM2RjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Iw6mYDpYo-_xQTVQb8Ts9qVDob6E9xCuCfGTaBtzOhA)
> 
> This will make it super easy:
> 
> https://desktop.github.com/
> 
> It's also available for Ubuntu https://github.com/shiftkey/desktop?tab=readme-ov-file#installation-via-package-manager
> 
> @dr-dolomite I really think it would be nice to see the option to switch languages on the fly. If need be we can restructure the www folder to allow this.
> 
> @snjzb I think I remember seeing something about switching character sets with an AT command, maybe that will help enable sending text with Chinese letters. I'll see if I can find it and I'll reply with it.
> 
> For now, your implementation of sending texts with simpleadmin would be a great contribution to the project. If you need help setting up GitHub for collaboration let me know! GitHub desktop will watch the repo folder and allow you to manage changes and pull requests into other repos on GitHub.

Maybe we can create a new branch for Chinese version and make it an option when installing Simple Admin from toolkit?

**Reply by snjzb** [reply]
I have translated Simple Admin and RMxxx_rgmii_toolkit into Chinese versions. Some features can be installed offline, but OPKG still requires a network.
![@~{3BP4D@GB_3`%M9CNPK{7](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/259c92ed-6301-4e24-ab5b-912611a95bbe)

**Reply by snjzb** [reply]
NEW SMS.ZIP
[SMS.zip](https://github.com/iamromulan/quectel-rgmii-toolkit/files/15442127/SMS.zip)
![R{ C3_$NCL 5MPE76%VA (Y](https://github.com/iamromulan/quectel-rgmii-toolkit/assets/170070306/686eaa45-6119-4d08-af3f-5e45645d06e4)
